tests/udp_socket.rs

```diff
@@ -757,16 +757,18 @@ fn send_recv_udp(mut tx: UdpSocket, mut rx: UdpSocket, connected: bool) {
             if event.is_readable() {
                 if let LISTENER = event.token() {
                     debug!("We are receiving a datagram now...");
-                    let cnt = unsafe {
-                        if !handler.connected {
-                            handler.rx.recv_from(handler.rx_buf.bytes_mut()).unwrap().0
-                        } else {
-                            handler.rx.recv(handler.rx_buf.bytes_mut()).unwrap()
-                        }
+                    let data = &mut handler.rx_buf.bytes_mut();
+                    let data = unsafe {
```

Comment:
Why not use the [`AsMut`](https://docs.rs/bytes/0.5.3/bytes/struct.BytesMut.html#impl-AsMut%3C%5Bu8%5D%3E) implementation?

Reply:
I tried with `let data = &mut rx_buf.as_mut()` but `rx_buf.as_ref()` returns empty so later assertion is failed.

Reply:
I tried the code locally and I see what you mean.

However this code isn't what the old `bytes_mut` did. The old `bytes_mut` returned a slice of the remaining capacity, this returns a slice to the already used bytes.